### DISCUSSION TITLE
Input images of different scales but output the same GFLOPs

### DISCUSSION THREAD

**Discussion by Lonely127** [ultralytics/ultralytics#8438] [Q&A]
Recently, when I was training with yolov8s, the input image sizes were different (such as 640, 1280, etc.), but the output GFLOPs were the same. I would like to ask you how to deal with this, and what needs to be changed. I have been working on it for a long time and haven't gotten it right. I hope you can get it. Boss help.

**Comment by pderrenger** [comment]
Hi there! 👋 The GFLOPs reported are typically calculated based on a single input size, which is often the default size used during the model's creation or benchmarking. If you're seeing the same GFLOPs for different input sizes, it's possible that the reported GFLOPs are not being recalculated for each input size dynamically.

To address this, you can manually calculate the GFLOPs for each input size using the model's `.info()` method after setting the desired image size. Here's a quick example in Python:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8s.pt')

# Set to desired image size and calculate GFLOPs
img_sizes = [640, 1280]
for img_size in img_sizes:
    model.imgsz = img_size
    info = model.info()
    print(f"GFLOPs for imgsz {img_size}: {info['gflops']}")
```

This should give you the correct GFLOPs for each specified input size. Let me know if this helps or if you have any more questions! 😊
